nifi-nar-bundles/nifi-dropbox-bundle/nifi-dropbox-processors/src/main/java/org/apache/nifi/processors/dropbox/ListDropbox.java

```diff
@@ -0,0 +1,277 @@
+import org.apache.nifi.components.state.Scope;
+import org.apache.nifi.context.PropertyContext;
+import org.apache.nifi.dropbox.credentials.service.DropboxCredentialDetails;
+import org.apache.nifi.dropbox.credentials.service.DropboxCredentialService;
+import org.apache.nifi.expression.ExpressionLanguageScope;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.processor.util.list.AbstractListProcessor;
+import org.apache.nifi.processor.util.list.ListedEntityTracker;
+import org.apache.nifi.serialization.record.RecordSchema;
+
+@PrimaryNodeOnly
+@TriggerSerially
+@Tags({"dropbox", "storage"})
+@CapabilityDescription("Retrieves a listing of files from Dropbox." +
+        " Each listed file may result in one FlowFile, the metadata being written as Flowfile attributes." +
+        " When the 'Record Writer' property is set - the entire result is written as records to a single FlowFile." +
+        " This Processor is designed to run on Primary Node only in a cluster. If the primary node changes, the new Primary Node will pick up where the" +
+        " previous node left off without duplicating all of the data.")
+@InputRequirement(Requirement.INPUT_FORBIDDEN)
+@WritesAttributes({@WritesAttribute(attribute = DropboxFileInfo.ID, description = "The Dropbox identifier of the file"),
+        @WritesAttribute(attribute = DropboxFileInfo.PATH, description = "The folder path where the file is located"),
+        @WritesAttribute(attribute = DropboxFileInfo.FILENAME, description = "The name of the file"),
+        @WritesAttribute(attribute = DropboxFileInfo.SIZE, description = "The size of the file"),
+        @WritesAttribute(attribute = DropboxFileInfo.TIMESTAMP, description = "The server modified time, when the file was uploaded to Dropbox"),
+        @WritesAttribute(attribute = DropboxFileInfo.REVISION, description = "Revision of the file")})
+@Stateful(scopes = {Scope.CLUSTER}, description = "The processor stores necessary data to be able to keep track what files have been listed already. " +
+        "What exactly needs to be stored depends on the 'Listing Strategy'.")
+public class ListDropbox extends AbstractListProcessor<DropboxFileInfo>  {
+
+    public static final PropertyDescriptor FOLDER_NAME = new PropertyDescriptor.Builder()
+            .name("folder-name")
+            .displayName("Folder Name")
+            .description("The name of the folder from which to pull list of files."+
+                    " Providing empty string as folder lists files from user root directory." +
+                    " Folder name should match the following regular expression pattern: (/(.|[\\r\\n])*)?|id:.*|(ns:[0-9]+(/.*)?) ")
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .addValidator(StandardValidators.createRegexMatchingValidator(Pattern.compile("(/(.|[\\r\\n])*)?|id:.*|(ns:[0-9]+(/.*)?)")))
+            .build();
```

Comment:
I found some issues with this property:
- if I leave it empty (null) I get an error
- if I specify `/` I get an error
- root is empty string (which is not really common config in NiFi), on the other hand, folders need to be specified with leading `/`

I would suggest using null value for the root directory (even if the API requires empty string).
Also not requiring leading `/` for the directories (which is more consistent with the root directory notation, either is null or empty string).

The regular expression seems quite hard to understand. Could you please also add some examples and/or description on the format?